python/ptest/cases/gomspace_checkout_case.py

```diff
@@ -133,29 +133,36 @@ def write_state(self, string_state, state_value):
         if ppt_mode_cmd == ppt_mode_updated:
             print("Could not update pptmode")
 
-        heater_cmd = self.str_to_bool(read_state(self, "gomspace.heater_cmd"))
-        heater_cmd_updated = self.str_to_bool(write_state(
-            self, "gomspace.heater_cmd", not heater_cmd))
-        if heater_cmd == heater_cmd_updated:
-            print("Could not update heater")
+        heater_init = self.str_to_bool(read_state(self, "gomspace.heater"))
+        write_state(self, "gomspace.heater_cmd", str(not heater_init).lower())
+        heater_updated=self.str_to_bool(read_state(self, "gomspace.heater"))
+
+        cycle_init = int(read_state(self, "pan.cycle_no"))
+        cycle_no = int(read_state(self, "pan.cycle_no"))
+        while(cycle_no-cycle_init<100 and heater_init==heater_updated):
+            cycle_no = int(read_state(self, "pan.cycle_no"))
+            heater_updated=self.str_to_bool(read_state(self, "gomspace.heater"))
+
+        if (heater_init==heater_updated):
+            print("could not update heater")
 
         counter_reset_cmd = self.str_to_bool(read_state(
             self, "gomspace.counter_reset_cmd"))
         counter_reset_cmd_updated = self.str_to_bool(write_state(
-            self, "gomspace.counter_reset_cmd", not counter_reset_cmd))
+            self, "gomspace.counter_reset_cmd", str(not counter_reset_cmd).lower())
         if counter_reset_cmd == counter_reset_cmd_updated:
             print("Could not update counter_reset")
 
         gs_reset_cmd = self.str_to_bool(
             read_state(self, "gomspace.gs_reset_cmd"))
         gs_reset_cmd_updated = self.str_to_bool(write_state(
-            self, "gomspace.gs_reset_cmd", not gs_reset_cmd))
+            self, "gomspace.gs_reset_cmd", str(not gs_reset_cmd).lower())
```

Comment:
Same question here, shouldn't we test that things got reset? Like the PPT mode

Reply:
See previous reply